api/v1alpha3/awscluster_types.go

```diff
@@ -68,6 +68,10 @@ type AWSClusterStatus struct {
 // +kubebuilder:resource:path=awsclusters,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="Cluster infrastructure is ready for EC2 instances"
+// +kubebuilder:printcolumn:name="VPC",type="string",JSONPath=".spec.networkSpec.vpc.id",description="AWS VPC the cluster is using"
+// +kubebuilder:printcolumn:name="Endpoint",type="string",JSONPath=".status.apiEndpoints[0]",description="API Endpoint",priority=1
+// +kubebuilder:printcolumn:name="Bastion IP",type="string",JSONPath=".status.bastion.publicIp",description="Bastion IP address for breakglass access"
```

Comment:
Rename "Name" to Cluster, or do exactly the same? Are there instances where AWSCluster name != Cluster name?

Reply:
The most reliable method might be to use the ownerRef similar to `AWSMachine` -> `Machine`, there is nothing that would enforce that the `awsCluster.Metadata.Name == `machine.Metadata.Name`

Reply:
That's what I wanted to do, but it's not accessible as a printer column unfortunately.

Reply:
More accurately, ownerReferences is an array, and the printer column subsystem doesn't support jsonpath queries, which is what I wanted to do, i.e. pick the ownerreference which is of the cluster type, and give the name.
Think I can do indexing into the array. So the question is: is ownerReference at index 0 always going to be the cluster?

Reply:
Probably, but we shouldn't rely on it

Reply:
We can do this in v1alpha3 instead, where Spec.ClusterName is going to be required

Reply:
@vincepri are we making that a requirement for providers? I thought we were only making that a requirement on `Machine`

Reply:
Ah that's true. Not sure if it's possible to add to the printer column, but the cluster-name label is going to be set by the Machine controller on external objects

Reply:
+1 to using the cluster-name label, then

Reply:
Already doing that for Machine. Should I do anything for AWSCluster then?

Reply:
We could add it there as well, but it'd look kinda weird to have a cluster name different from the AWSCluster / confuse folks

Reply:
That's what I thought too.
Going to leave the PR as is then. Up/down vote as appropriate.

Reply:
With that rationale, should we remove the Machine column from the awsmachine_types below as well? Since `awsMachine.Metadata.Name` is not guaranteed to be equal to `machine.Metadata.Name`?

Reply:
I deliberately added that so people can cross-reference. There's been some confusion on Slack about not being able to find the matching resources. Because the machine ref isn't in an array, I am actually able to pull it out.

Reply:
Would folks not also want to cross reference the `AWSCluster` to the associated `Cluster` as well, then?

Reply:
OK. Added.